.github/workflows/python-package.yml

```diff
@@ -57,5 +57,32 @@ jobs:
         ruff check . --exit-zero --line-length=127 --statistics
     - name: Test with pytest
       run: |
-        if [[ ! -e build ]]; then mkdir build; fi
-        PYTHONPATH=. pytest -vvv davidia/tests/
+        pytest -vvv davidia/tests/
+    - name: Build sdist and wheel
+      run: pipx run build davidia
+    - name: Upload sdist and wheel as artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: dist
+        path: dist
+
+  pypi:
+    # upload to PyPI and make a release on every tag
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
```

Comment:
Also need to add `tags` to `on.push` otherwise we need to publish on `main`

Reply:
Added